### PR TITLE
Adding a new ordering based on identifying duplicate reads: SAMRecordDup...

### PR DESCRIPTION
**Pull request by nh13**
...licateComparator.

Adding an ability to iterate through duplicate sets using a DuplicateSetIterator.

### CODE REVIEW

**Comment by yfarjoun** [comment]
This is AWESOME!

File: src/java/htsjdk/samtools/SAMRecordDuplicateComparator.java [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+    /** Is this paired end and are both ends of a pair mapped */
+    private boolean pairedEndAndBothMapped(final SAMRecord record) {
+        return (record.getReadPairedFlag() && !record.getReadUnmappedFlag() && !record.getMateUnmappedFlag());
+        
+    }
+
+    /**
+     * Most stringent comparison.
+     *
+     * * Two records are compared based on if they are duplicates of each other, and then based
```
Comment (nh13) [inline src/java/htsjdk/samtools/SAMRecordDuplicateComparator.java]: ```* *``` -> ```*```

**Comment by nh13** [comment]
@yfarjoun since I will be out, do you want to champion the code review and inclusion into htsjdk with the team over the next sprint?

**Comment by yfarjoun** [comment]
I could try...I guess you're not going to make it before the end of the
week...

Y.

**Comment by gbggrant** [comment]
Looks good.  :+1:

**Comment by vdauwera** [comment]
@yfarjoun Can this be merged since @gbggrant gave the :+1: ? I ask you since you were the official designated champion of this PR.

**Pull request merged**